src/Ellipsis.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component, createRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { throttle } from 'lodash';
+
+class Ellipsis extends Component {
+    containerRef = createRef();
+
+    state = {
+        isPrepared: false,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.throttleHandleResize = throttle(this.handleResize, 250);
+    }
+
+    componentDidMount() {
+        window.addEventListener('resize', this.throttleHandleResize, { passive: true });
+
+        this.ellipsize();
+
+        this.setState({ isPrepared: true });
+    }
+
+    componentDidUpdate() {
+        this.ellipsize();
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.throttleHandleResize, { passive: true });
+    }
+
+    render() {
+        const { children, tag: Tag, className } = this.props;
+        const { isPrepared } = this.state;
+
+        const containerClassName = classNames('container', {
```

Comment:
Let's follow our guideline of `package-name_class` to try to avoid classes collisions with other packages:

```suggestion
        const containerClassName = classNames('react-ellipsis_container', {
```